js/deribit.js

```diff
@@ -458,28 +458,66 @@ module.exports = class deribit extends Exchange {
                 const id = this.safeString (market, 'instrument_name');
                 const baseId = this.safeString (market, 'base_currency');
                 const quoteId = this.safeString (market, 'quote_currency');
+                const settleId = quoteId;
                 const base = this.safeCurrencyCode (baseId);
                 const quote = this.safeCurrencyCode (quoteId);
+                const settle = this.safeCurrencyCode (settleId);
                 const type = this.safeString (market, 'kind');
-                const future = (type === 'future');
+                const settlementPeriod = this.safeValue (market, 'settlement_period');
+                const swap = (settlementPeriod === 'perpetual');
+                const future = !swap && (type === 'future');
                 const option = (type === 'option');
-                const active = this.safeValue (market, 'is_active');
+                let symbol = quote + '/' + base + ':' + settle;
+                let expiry = undefined;
+                let strike = undefined;
+                let optionType = undefined;
+                if (option || future) {
+                    expiry = this.yyyymmdd (this.iso8601 (this.safeInteger (market, 'expiration_timestamp')), '');
```

Comment:
`yyyymmdd` takes an integer, while `iso8601` returns a string, i think `iso8601` is not necessary here

Reply:
yep, this hasn't been tested...